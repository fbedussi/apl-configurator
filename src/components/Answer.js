import React from 'react';
import { connect } from 'react-redux';

import { getLabels, getBreadcrumbs, getAnswersHistory, getShowForm, getQuestions, getCurrentAnswer, getCurrentNode } from '../selectors';
import { goBack, toggleForm, submitForm } from '../actions';

import Title from './Title';
import Back from './Back';
import Images from './Images';
import Recap from './Recap';
import QuotationButton from './QuotationButton';
import Form from './Form';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    currentNode: getCurrentNode(state),
    breadcrumbs: getBreadcrumbs(state),
    answersHistory: getAnswersHistory(state),
    questions: getQuestions(state),
    showForm: getShowForm(state),
    currentAnswer: getCurrentAnswer(state)
});

const mapDispatchToProps = (dispatch) => ({
    goBack: (currentNode) => dispatch(goBack(currentNode)),
    toggleForm: () => dispatch(toggleForm()),
    submitForm: (data) => dispatch(submitForm(data))
});

class Answer extends React.Component {
    render() {
        const { currentAnswer, questions, labels, breadcrumbs, answersHistory, showForm, submitForm, currentNode } = this.props;

        const className = `slideInner answer ${showForm ? 'showForm' : 'hideForm'}`;

        return (
            <div className={className}>
                <Title text={currentAnswer.title} />
                {currentAnswer.subtitle && <p className="subtitle">{currentAnswer.subtitle}</p>}
                <p className="answer-text" dangerouslySetInnerHTML={{__html: currentAnswer.text}}/>
                <div className="answer-recapAndImage">
                    <Images
                        sources={currentAnswer.images}
                    />
                    <Recap
                        questions={questions}
                        labels={labels}
                        breadcrumbs={breadcrumbs}
                        answersHistory={answersHistory}
                    />
                </div>
                <Form
                    labels={labels}
                    answer={currentAnswer}
                    submitHandler={(e) => {
                        var form;
                        var data;

                        e.preventDefault();

                        form = e.target;
                        data = new FormData(form);

                        submitForm(data);
                    }}
                />

                <QuotationButton
                    show={!showForm}
                    text={labels.quotationBtn}
                    clickHandler={() => this.props.toggleForm()}
                />

                <Back
                    show={Boolean(breadcrumbs.length)}
                    label={labels.back}
                    clickHandler={() =>
                        this.props.goBack(currentNode)
                    }
                />
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Answer);
