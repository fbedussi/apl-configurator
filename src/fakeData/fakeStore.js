import texts from './apl-texts.json';

const fakeStore = {
    currentLanguage: texts.languageIsoCode,
    showForm: false,
    steps: 6,
    stepsLeft: 6,
    goingBack: false,
    breadcrumbs: [],
    answersHistory: [],
    labels: texts.labels,
    answers: texts.answers,
    questions: texts.questions,
    tree: texts.tree,
    currentNode: texts.tree,
    requestStatus: 'unsent' // || sent || success || error
};

export default fakeStore;
