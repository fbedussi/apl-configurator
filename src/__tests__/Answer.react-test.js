import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import Answer from '../components/Answer';
import renderer from 'react-test-renderer';
import fakeStore from '../fakeData/fakeStore';

fakeStore.currentNode = {
  "id": 52,
  "questionId": null,
  "answerId": 9
};
const store = createStore(() => fakeStore);

test('App component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Answer />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
