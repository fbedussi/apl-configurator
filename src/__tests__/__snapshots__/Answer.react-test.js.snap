// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div
  className="slideInner answer hideForm"
>
  <h1
    className="title"
  >
    APL Smart
  </h1>
  <p
    className="subtitle"
  >
    soluzione 2 o 3 con pulsante touch
  </p>
  <p
    className="answer-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
  <div
    className="answer-recapAndImage"
  >
    <div
      className="imagesContainer"
    >
      <img
        className="image"
        src="images/APL_Smart_soluzione2_con_pulsante_touch.jpg"
      />
      <img
        className="image"
        src="images/APL_Smart_soluzione3_con_pulsante_touch.jpg"
      />
    </div>
    <div
      className="recap"
    >
      <table
        className="recap-table"
      >
        <thead
          className="recap-head"
        >
          <tr>
            <th>
              domanda
            </th>
            <th>
              risposta
            </th>
          </tr>
        </thead>
        <tbody
          className="recap-body"
        />
      </table>
    </div>
  </div>
  <form
    className="quotation-form"
    name="contactForm"
    onSubmit={[Function]}
  >
    <p
      className="form-mandatory"
    >
      * campi obbligatori
    </p>
    <input
      id="solution"
      name="solution"
      type="hidden"
      value="APL Smart - soluzione 2 o 3 con pulsante touch"
    />
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        Nome
      </label>
      <input
        className="form-input"
        id="name"
        minLength={undefined}
        name="name"
        required={false}
        type="text"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="surname"
      >
        Cognome
      </label>
      <input
        className="form-input"
        id="surname"
        minLength={undefined}
        name="surname"
        required={false}
        type="text"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="company"
      >
        Societ√† (ragione sociale)*
      </label>
      <input
        className="form-input"
        id="company"
        minLength={undefined}
        name="company"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="address"
      >
        Indirizzo*
      </label>
      <input
        className="form-input"
        id="address"
        minLength={undefined}
        name="address"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        E-mail*
      </label>
      <input
        className="form-input"
        id="email"
        minLength={undefined}
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="phone"
      >
        Telefono
      </label>
      <input
        className="form-input"
        id="phone"
        minLength={undefined}
        name="phone"
        required={false}
        type="tel"
      />
    </div>
    <div
      className="form-field"
    >
      <label
        className="form-label"
        htmlFor="website"
      >
        Sito
      </label>
      <input
        className="form-input"
        id="website"
        minLength={undefined}
        name="website"
        required={false}
        type="text"
      />
    </div>
    <div
      className="form-field"
    >
      <span
        className="form-label"
      >
        Motivo della richiesta
        *
      </span>
      <fieldset
        className="form-input form-reason"
      >
        <span
          className="form-reason-option"
        >
          <input
            id="project"
            name="reason"
            required={true}
            type="radio"
            value="project"
          />
          <label
            htmlFor="project"
          >
            Elaborazione progetto
          </label>
        </span>
        <span
          className="form-reason-option"
        >
          <input
            id="call"
            name="reason"
            required={true}
            type="radio"
            value="call"
          />
          <label
            htmlFor="call"
          >
            Miglioria per gara d'appalto
          </label>
        </span>
        <span
          className="form-reason-option"
        >
          <input
            id="other"
            name="reason"
            required={true}
            type="radio"
            value="other"
          />
          <label
            htmlFor="other"
          >
            Altro
          </label>
        </span>
      </fieldset>
    </div>
    <label
      className="form-field"
    >
      <span
        className="form-label"
      >
        Eventuali informazioni aggiuntive
      </span>
      <textarea
        className="form-msg form-input"
        id="msg"
        name="msg"
      />
    </label>
    <button
      className="ctaBtn form-submit"
      type="submit"
    >
      Invia
    </button>
  </form>
  <button
    className="quotation-btn ctaBtn ctaBtn--inverse"
    onClick={[Function]}
  >
    scopri quanto costa
  </button>
</div>
`;
